README.md

```diff
@@ -19,44 +19,80 @@ We assume that you already have:
 
 Then, you can create an URS for your circuit in the following way:
 
-```rust,ignore
-use kimchi::{circuits::constraints, verifier::verify};
-use mina_curves::pasta::{fp::Fp, vesta::{Affine, VestaParameters}, pallas::Affine as Other};
+```rust
+use std::sync::Arc;
+use ark_ff::{FftField, PrimeField};
+use ark_poly::{EvaluationDomain, Radix2EvaluationDomain as D};
+use circuit_construction::*;
+use commitment_dlog::{commitment::CommitmentCurve, srs::SRS};
+use groupmap::GroupMap;
+use kimchi::verifier::verify;
+use mina_curves::pasta::{
+    fp::Fp,
+    vesta::{Affine, VestaParameters},
+};
 use oracle::{
-    constants::PlonkSpongeConstantsKimchi,
+    constants::*,
     sponge::{DefaultFqSponge, DefaultFrSponge},
 };
-use commitment_dlog::commitment::{b_poly_coefficients, ceil_log2, CommitmentCurve};
-
-type SpongeParams = PlonkSpongeConstantsKimchi;
-type BaseSponge = DefaultFqSponge<VestaParameters, SpongeParams>;
-type ScalarSponge = DefaultFrSponge<Fp, SpongeParams>;
-
-// compile the circuit
-let fp_sponge_params = oracle::pasta::fp_kimchi::params();
-let cs = ConstraintSystem::<Fp>::create(gates, vec![], fp_sponge_params, public_size).unwrap();
-
-// create an URS
-let mut urs = SRS::<Affine>::create(cs.domain.d1.size as usize);
-srs.add_lagrange_basis(cs.domain.d1);
-
-// obtain a prover index
-let prover_index = {
-    let fq_sponge_params = oracle::pasta::fq_kimchi::params();
-    let (endo_q, _endo_r) = endos::<Other>();
-    Index::<Affine>::create(cs, fq_sponge_params, endo_q, srs)
+type SpongeQ = DefaultFqSponge<VestaParameters, PlonkSpongeConstantsKimchi>;
+type SpongeR = DefaultFrSponge<Fp, PlonkSpongeConstantsKimchi>;
+
+pub fn circuit<
+    F: PrimeField + FftField,
+    Sys: Cs<F>,
+>(
+    witness: Option<F>,
+    sys: &mut Sys,
+    public_input: Vec<Var<F>>,
+) {
+    let three = sys.constant(3u32.into());
+    let first_public_input = public_input[0];
+    let witness = sys.var(|| witness.unwrap());
+
+    //TODO replace the following free variable once the "add" gate constraint is ready to use.
+    let result = sys.var(|| {
+        first_public_input.val() + witness.val()
+    });
+    sys.assert_eq(result, three);
+}
+
+// create SRS
+let srs = {
+    //TODO how to determine depth value? it throws error when the depth is too large
```

Comment:
Yeah basically lol, the usecase we’d want to support is either 1) the user doesn’t care and we can generate the correct srs for them or 2) the user creates a large srs and this will be the upperbound for all circuits that use it (that’s what mina does)